tests/api_connexion/endpoints/test_variable_endpoint.py

```diff
@@ -29,38 +31,141 @@ def setUpClass(cls) -> None:
 
     def setUp(self) -> None:
         self.client = self.app.test_client()  # type:ignore
+        clear_db_variables()
+
+    def tearDown(self) -> None:
+        clear_db_variables()
 
 
 class TestDeleteVariable(TestVariableEndpoint):
-    @pytest.mark.skip(reason="Not implemented yet")
-    def test_should_response_200(self):
-        response = self.client.delete("/api/v1/variables/TEST_VARIABLE_KEY")
+    def test_should_response_204(self):
+        Variable.set("delete_var1", 1)
+        # make sure variable is added
+        response = self.client.get("/api/v1/variables/delete_var1")
+        assert response.status_code == 200
+
+        response = self.client.delete("/api/v1/variables/delete_var1")
         assert response.status_code == 204
 
+        # make sure variable is deleted
+        response = self.client.get("/api/v1/variables/delete_var1")
+        assert response.status_code == 404
+
 
 class TestGetVariable(TestVariableEndpoint):
-    @pytest.mark.skip(reason="Not implemented yet")
+
     def test_should_response_200(self):
+        expected_value = '{"foo": 1}'
+        Variable.set("TEST_VARIABLE_KEY", expected_value)
         response = self.client.get("/api/v1/variables/TEST_VARIABLE_KEY")
         assert response.status_code == 200
+        assert response.json == {"key": "TEST_VARIABLE_KEY", "value": expected_value}
+
+    def test_should_response_404_if_not_found(self):
+        response = self.client.get("/api/v1/variables/NONEXIST_VARIABLE_KEY")
+        assert response.status_code == 404
 
 
 class TestGetVariables(TestVariableEndpoint):
-    @pytest.mark.skip(reason="Not implemented yet")
-    def test_should_response_200(self):
+    @parameterized.expand([
+        ("/api/v1/variables?limit=2&offset=0", {
+            "variables": [
+                {"key": "var1"},
+                {"key": "var2"},
+            ],
+            "total_entries": 2,
+        }),
+        ("/api/v1/variables?limit=2&offset=1", {
+            "variables": [
+                {"key": "var3"},
+            ],
+            "total_entries": 1,
```

Comment:
```suggestion
            "variables": [
                {"key": "var2"},
                {"key": "var3"}
            ],
            "total_entries": 2,
```
The total data is 3, the limit is 2 and offset is 1. What do you think?

Reply:
limit is 2 means page size is 2, if we are at second page (offset=1), shouldn't it be just return the last variable?

Reply:
The way I understand this is that the total number of pages it should return is 2. If we have [1,2,3,4,5] items, it should return only 2 items. Now if we offset by 1, then it starts counting from 2 above and return maximum of 2 items. This is how I think I understand it. Would appreciate if you can explain with the above list.

Reply:
I see where the confusion is coming from. In the API spec, the parameter is named as `PageOffset`, which sounds like offset is the page number, not the absolute element count. But the description is set to `The number of items to skip before starting to collect the result set`.

I will change it to match the description in API spec.